index.js

```diff
@@ -1,2 +1,94 @@
+// Load environment variables
+require('dotenv').config();
+
+const slackEventsApi = require('@slack/events-api');
+const passport = require('passport');
+const SlackStrategy = require('@aoberoi/passport-slack').default.Strategy;
+const http = require('http');
 const express = require('express');
-const slack = require('slack');
\ No newline at end of file
+const bodyParser = require('body-parser');
+
+const { getClientByTeamId } = require('./util/getClient');
+
+const port = process.env.APP_PORT || 3000;
+
+// Initialize a data structures to store team authorization info (typically stored in a database)
+const botAuthorizations = {}
+
+// *** Initialize event adapter using verification token from environment variables ***
+const slackEvents = slackEventsApi.createSlackEventAdapter(
+  process.env.SLACK_VERIFICATION_TOKEN, 
+  { includeBody: true }
+);
+
+// Initialize Add to Slack (OAuth) helpers
+passport.use(new SlackStrategy({
+  clientID: process.env.SLACK_CLIENT_ID,
+  clientSecret: process.env.SLACK_CLIENT_SECRET,
+  skipUserProfile: true,
+}, (accessToken, scopes, team, extra, profiles, done) => {
+  botAuthorizations[team.id] = extra.bot.accessToken;
+  done(null, {});
+}));
+
+// Initialize an Express application
+const app = express();
+app.use(bodyParser.json());
+
+// Plug the Add to Slack (OAuth) helpers into the express app
+app.use(passport.initialize());
+app.get('/', (req, res) => {
+  res.send('<a href="/auth/slack"><img alt="Add to Slack" height="40" width="139" src="https://platform.slack-edge.com/img/add_to_slack.png" srcset="https://platform.slack-edge.com/img/add_to_slack.png 1x, https://platform.slack-edge.com/img/add_to_slack@2x.png 2x" /></a>');
+});
+app.get('/auth/slack', passport.authenticate('slack', {
+  scope: ['bot']
+}));
+app.get('/auth/slack/callback',
+  passport.authenticate('slack', { session: false }),
+  (req, res) => {
+    res.send('<p>Greet and React was successfully installed on your team.</p>');
+  },
+  (err, req, res, next) => {
+    res.status(500).send(`<p>Greet and React failed to install</p> <pre>${err}</pre>`);
+  }
+);
+
+// *** Plug the event adapter into the express app as middleware ***
+app.use('/slack/events', slackEvents.expressMiddleware());
+
+// *** Attach listeners to the event adapter ***
+
+// *** Greeting any user that says "hi" ***
+slackEvents.on('message', (message, body) => {
+  // Only deal with messages that have no subtype (plain messages) and contain 'hi'
+  if (!message.subtype && message.text.indexOf('hi') >= 0) {
```

Comment:
Can we make this check more strict by requiring the message to be exclusive "hi"? We are getting too many dev messages currently